src/Microsoft.AspNet.Routing/Template/TemplateRoute.cs

```diff
@@ -203,19 +203,31 @@ public virtual string GetVirtualPath(VirtualPathContext context)
             // Validate that the target can accept these values.
             var childContext = CreateChildVirtualPathContext(context, values.AcceptedValues);
 
-            var path = _target.GetVirtualPath(childContext);
-            if (path != null)
+            var pathData = _target.GetVirtualPath(childContext);
+            if (pathData != null)
             {
                 // If the target generates a value then that can short circuit.
-                return path;
+                return pathData;
             }
 
             // If we can produce a value go ahead and do it, the caller can check context.IsBound
             // to see if the values were validated.
-            path = _binder.BindValues(values.AcceptedValues);
+            var tempDataTokens = new RouteValueDictionary();
+            if (DataTokens != null)
+            {
+                foreach (var dataToken in DataTokens)
+                {
+                    tempDataTokens.Add(dataToken.Key, dataToken.Value);
+                }
+            }
```

Comment:
@rynowak 
It seems not. I found that from one of the constructors of RouteVauleDictionary, but it takes IDictionary<string, object> instead of IReadOnlyDictionary. Is there any thing you have in mind??